### PR TITLE
Add field grouping ScalaDoc for ArbiterIO

### PR DESCRIPTION
**Pull request by Burnleydev1**
### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement
Documentation

#### API Impact
Unknown Impact

#### Backend Code Generation Impact
No impact

#### Desired Merge Strategy
Squash

#### Release Notes
Improved documentation for ArbiterIO

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.3.x`, [small] API extension: `3.4.x`, API modification or big change: `3.5.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

**Comment by Burnleydev1** [comment]
Hello @mwachs5, Thanks for the review and corrections I committed the suggestions.

**Comment by Burnleydev1** [comment]
Hello @mwachs5, please I wish to what the way forward with these pull requests is: #2208 and #2214 is

File: src/main/scala/chisel3/util/Arbiter.scala [hunk 1]
```diff
@@ -10,15 +10,28 @@ import chisel3.internal.naming.chiselName  // can't use chisel3_ version because
...
 /** IO bundle definition for an Arbiter, which takes some number of ready-valid inputs and outputs
   * (selects) at most one.
+  * @groupdesc Signals The actual hardware fields of the Bundle
   *
   * @param gen data type
   * @param n number of inputs
   */
 class ArbiterIO[T <: Data](private val gen: T, val n: Int) extends Bundle {
-  // See github.com/freechipsproject/chisel3/issues/765 for why gen is a private val and proposed replacement APIs.
+ //See github.com/freechipsproject/chisel3/issues/765 for why gen is a private val and proposed replacement APIs.
```
Comment (mwachs5) [inline src/main/scala/chisel3/util/Arbiter.scala:19]:
```suggestion
   // See github.com/freechipsproject/chisel3/issues/765 for why gen is a private val and proposed replacement APIs.
```

basically I do not want to see a diff on this line

**Comment by Burnleydev1** [comment]
Hi @mwachs5, I went through it to check of syntax or ScalaDoc errors, and I think including your commit suggestions is no longer different from that on the master branch.

**Review by mwachs5 — APPROVED** [review]

**Pull request merged**